src/compose.ts

```diff
@@ -62,12 +62,10 @@ export function normalize(c: any): Composition {
 			// Normalise services
 			const services: Dict<Service> = c.services;
 			const serviceNames = _.keys(services);
-			if (!_.size(services)) {
-				throw new ValidationError('Composition defines no services');
-			}
+
 			c.services = _.mapValues(services, (service) => {
 				return normalizeService(service, serviceNames);
-			});
+			}) || { };
```

Comment:
Why is that needed? If it can be undefined, let's set it above, at `const services = c.services || {}`.

Reply:
I guess it does make more sense there, you're right.